data/schema/game_config.cfg

```diff
@@ -72,16 +72,16 @@
             [/type]
         [/union]
     [/type]
+    # definition of range_list and s_range_list, before macros are expanded
     {LIST_TYPE_COMPLEX range (
         [element]
-            [list]
-                split="-"
-                min=1
-                max=2
-                [element]
-                    link=unsigned
-                [/element]
-            [/list]
+            value="\d+"
+        [/element]
+        [element]
+            value="\d+-\d+"
+        [/element]
+        [element]
+            value="\d+-infinity"
```

Comment:
...wait, aren't those first two `[element]` tags identical to the previously-existing `[element]` tag? Why'd you have to split it up like that?

Your commit message implies the schema was accepting negative numbers, but if the code on the left accepts negative numbers, I think that might be a bug in the schema implementation... the only way I could see it accepting negative numbers is if `unsigned` accepts an empty string.

Reply:
IIUC, `[list]` accepts empty elements, as long as the number of elements that match is between min and max. So "-7" is split to "", "7", which gets one match and passes the test.

https://github.com/wesnoth/wesnoth/blob/b08677c67807a6ff2db34fcb5ebafb7f35d8ad36/src/serialization/schema/type.cpp#L115-L116

I tried changing that "true" to "false", and the validation complained about the blank lines not matching map_data in:
https://github.com/wesnoth/wesnoth/blob/b08677c67807a6ff2db34fcb5ebafb7f35d8ad36/data/scenario-test.cfg#L3127-L3138

Reply:
Hmm... I'm not sure what's best here... obviously the map data needs to match, but I don't think skipping empty lines is even quite the right approach there (is a blank line in the middle of the map actually valid?)... and I think for most lists not splitting on line breaks, empty elements are undesirable.